server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMessageListMethod.java

```diff
@@ -103,66 +86,39 @@
     public Stream<JmapResponse> process(JmapRequest request, ClientId clientId, MailboxSession mailboxSession) {
         Preconditions.checkArgument(request instanceof GetMessageListRequest);
         GetMessageListRequest messageListRequest = (GetMessageListRequest) request;
-        GetMessageListResponse messageListResponse = getMessageListResponse(messageListRequest, clientId, mailboxSession);
+        GetMessageListResponse messageListResponse = getMessageListResponse(messageListRequest, mailboxSession);
  
         Stream<JmapResponse> jmapResponse = Stream.of(JmapResponse.builder().clientId(clientId)
                 .response(messageListResponse)
                 .responseName(RESPONSE_NAME)
                 .build());
-        return Stream.<JmapResponse> concat(jmapResponse, 
+        return Stream.concat(jmapResponse,
                 processGetMessages(messageListRequest, messageListResponse, clientId, mailboxSession));
     }
 
-    private GetMessageListResponse getMessageListResponse(GetMessageListRequest messageListRequest, ClientId clientId, MailboxSession mailboxSession) {
+    private GetMessageListResponse getMessageListResponse(GetMessageListRequest messageListRequest, MailboxSession mailboxSession) {
         GetMessageListResponse.Builder builder = GetMessageListResponse.builder();
         try {
             MultimailboxesSearchQuery searchQuery = convertToSearchQuery(messageListRequest);
-            Map<MailboxId, Collection<MessageUid>> searchResults = mailboxManager.search(searchQuery, mailboxSession);
-            
-            aggregateResults(mailboxSession, searchResults).entries().stream()
-                .sorted(comparatorFor(messageListRequest))
-                .map(entry -> entry.getValue().getMessageId())
+            mailboxManager.search(searchQuery,
+                mailboxSession,
+                messageListRequest.getLimit().orElse(maximumLimit) + messageListRequest.getPosition())
+                .stream()
                 .skip(messageListRequest.getPosition())
```

Comment:
Are you sure I'm wrong? I don't see any test regarding skip/position handling.
Please add a test, and keep like this if it's working, or handling it correctly if it's not working.
By correctly I don't mean to necessarily implement skip if it is complicated, it can be to disallow position requests (of course if they are not needed by inbox)